src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -427,20 +426,27 @@ protected void Stop()
 
                     return;
                 }
+            }
+            else if (customPipeNameIsSet)
+            {
+                if (runspaceVersion.Version.Major < 6 && runspaceVersion.Version.Minor < 2)
+                {
+                    await requestContext.SendErrorAsync(
+                        $"Attaching to a process with CustomPipeName is only available with PowerShell 6.2 and higher (current session is {runspaceVersion.Version}).");
+                    return;
+                }
 
-                // Clear any existing breakpoints before proceeding
-                await ClearSessionBreakpointsAsync();
-
-                // Execute the Debug-Runspace command but don't await it because it
-                // will block the debug adapter initialization process.  The
-                // InitializedEvent will be sent as soon as the RunspaceChanged
-                // event gets fired with the attached runspace.
-                int runspaceId = attachParams.RunspaceId > 0 ? attachParams.RunspaceId : 1;
-                _waitingForAttach = true;
-                Task nonAwaitedTask =
-                    _editorSession.PowerShellContext
-                        .ExecuteScriptStringAsync($"\nDebug-Runspace -Id {runspaceId}")
-                        .ContinueWith(OnExecutionCompletedAsync);
+                await _editorSession.PowerShellContext.ExecuteScriptStringAsync(
+                    $"Enter-PSHostProcess -CustomPipeName {attachParams.CustomPipeName}",
+                    errorMessages);
+
+                if (errorMessages.Length > 0)
+                {
+                    await requestContext.SendErrorAsync(
+                        $"Could not attach to process with CustomPipeName: '{attachParams.CustomPipeName}'");
+
+                    return;
+                }
             }
             else
```

Comment:
Might be a good opportunity to take this out?

Reply:
Given they additional check to make sure processId is an int, refactoring to remove `else` just doesn't seem as readable than this.